### PR TITLE
Feature/ci issue 104

### PR DESCRIPTION
**Pull request by amanganiello90**

### CODE REVIEW

**Comment by mraible** [comment]
It's probably best to have @mshima or @pascalgrimaud review this one.

File: generators/ci-cd/templates/github-ci.yml.ejs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            <%_ if (!skipClient) { _%>
+            JHI_DISABLE_WEBPACK_LOGS: true
+            <%_ } _%>
+            <%_ if (clientFramework === ANGULAR) { _%>
+            NG_CLI_ANALYTICS: false
+            <%_ } _%>
+        steps:
+            - uses: actions/checkout@v2
+              <%_ if (!skipClient) { _%>
+            - uses: actions/setup-node@v1
```
Comment (mshima) [inline generators/ci-cd/templates/github-ci.yml.ejs:41]:
```suggestion
            - uses: actions/setup-node@v2
```

File: generators/ci-cd/templates/github-ci.yml.ejs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+              with:
+                  node-version: <%= NODE_VERSION %>
+              <%_ }
```
Comment (mshima) [inline generators/ci-cd/templates/github-ci.yml.ejs:44]:
```suggestion
            - uses: actions/setup-node@v1
              with:
                  node-version: <%= NODE_VERSION %>
```
Node is required for this blueprint?

File: generators/ci-cd/templates/github-ci.yml.ejs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+              if (!skipClient) { _%>
+            - name: Install node.js packages
+              run: <%= clientPackageManager %> install
+              <%_ }
+              if (!skipServer) { _%>
+            - name: Run backend test
+              run: |
+                  cd server
+                  <%= clientPackageManager %> install
+                  <%= clientPackageManager %> run test
```
Comment (mshima) [inline generators/ci-cd/templates/github-ci.yml.ejs:58]:
```suggestion
              run: |
                  <%= clientPackageManager %> install
                  <%= clientPackageManager %> run test
              working-directory: server
```
`cd` can be replaced with `working-directory`.

**Review by mshima — APPROVED** [review]
I could review the generator and GitHub actions.

I can't help with others files.

File: generators/ci-cd/templates/github-ci.yml.ejs (hunk 2 above)
Comment (amanganiello90) [reply generators/ci-cd/templates/github-ci.yml.ejs]: Yes, I have just committed the change

**Comment by amanganiello90** [comment]
Any news?

**Comment by mshima** [comment]
From my side it’s ok, it’s approved.
I just think you should try to add conditions and variables to upstream templates instead of copy the entire template.
Could make maintenance easier.

We probably won’t get a review of the pipelines. It’s too specific.
IMO just go ahead and merge.

**Comment by amanganiello90** [comment]
@mshima Thanks for the answer.
I don't know if it is possible to add variables in the templates, because the pipeline for a nodejs app is quite different from the standard jhipster ci-cd.

**Pull request merged**